### DISCUSSION TITLE
Support for LLRT

### DISCUSSION THREAD

**Discussion by caioquirino** [aws-powertools/powertools-lambda-typescript#2116] [Ideas]
Hi all, good evening!

I haven't tried powertools with LLRT yet, I am happy to know about any experience about it, and if some effort is necessary, I am happy to contribute! I will start testing it in the upcoming days as well :) Thanks!

**Comment by dreamorosi** [comment]
Hi, thanks for opening this discussion.

We have a feature request that was opened last week (#2050), but at the moment we are blocked by upstream limitations in LLRT.

At present none of the core utilities (Logger, Metrics, and Tracer) will work on LLRT due to several Node.js APIs missing or not usable in user code, such as: `Console`, `http`, `fs`, `assert`, `querystring`, etc.. We haven't tested any of the other Powertools utilities, but if you do please let us know.

Overall our position on LLRT can be summed up in what I shared in this comment to the issue I linked:
> As a side note, I want to remark that according to both our tenets ([see first point](https://docs.powertools.aws.dev/lambda/typescript/latest/#tenets)) and versioning policy (see [dependency lifecycle section](https://docs.powertools.aws.dev/lambda/typescript/latest/versioning/#dependency-lifecycle)) our current priority is to officially support AWS Lambda managed runtimes - aka `nodejs16x`, `nodejs18x`, and `nodejs20x` at the time of writing.

> LLRT has been met with remarkable positive feedback and shows a lot of promise, however while we are excited about it and eager to see where it goes, making significant changes to Powertools to support it would be premature since LLRT itself is considered experimental and subject to change.

> With that said, we are open to work with the LLRT team to fix this specific issue, and other similar ones where LLRT is missing support for common Node.js APIs. Likewise, we are open to consider making changes on our side only as long as they don't impact compatibility, DX, and performance with the main Node.js runtimes.

From our side, my only request would be to consider leaving a comment or upvoting [this feature](https://github.com/awslabs/llrt/issues/130) request on the LLRT repo would already unlock both `Metrics` and `Logger`.
